shell/platform/android/io/flutter/embedding/engine/systemchannels/TextInputChannel.java

```diff
@@ -112,6 +113,22 @@ public void onMethodCall(@NonNull MethodCall call, @NonNull MethodChannel.Result
               textInputMethodHandler.clearClient();
               result.success(null);
               break;
+            case "TextInput.sendAppPrivateCommand":
+              try {
+                final JSONObject arguments = (JSONObject) args;
+                final String action = arguments.getString("action");
+                final String data = arguments.getString("data");
+                Bundle bundle = null;
+                if (data != null && !data.isEmpty()) {
```

Comment:
I don't really know how this works. Do you want to check action too (as a generic mechanism besides gboard)?

Reply:
The Json args for the Crowdsource project look like this:
{
  ‘action’: 'com.google.android.apps.village.boond.GET_INPUT_CONTEXT',
  ‘data’: ‘’
}

I don't think we need to check the action. If action is not a string, a json error would be thrown or if it is an empty string, the Android platform should handle it. Data is checked in here because it affects if we need to pass a Bundle object or a null.  It also follows the same pattern like how the "TextInput.setEditableSizeAndTransform" being process in this function.